### PR TITLE
Feature/sc 109246/implement move a folder to another collection

### PR DESCRIPTION
**Pull request by markhornak**
Overview
Shortcut: [109246](https://app.shortcut.com/guru/story/109246/implement-move-a-folder-to-another-collection)
this provides the ability to move a folder to a different collection

added `move_folder_to_collection` method in `core.py`
added `move_to_collection` helper method for Folder object in `data_objects.py'

### CODE REVIEW

File: guru/core.py [hunk 1]
```diff
@@ -2614,6 +2614,64 @@ def add_card_to_folder(self, card, target_folder):
...
+                    its new collection. This helpful if you want to do multiple operations, like move
+                    a folder to a new collection then add it to a folder group there. By default this is
+                    0 so it doesn't wait. If you set a timeout of 10, we'll wait up to 10 seconds to
+                    see if the move completes.
+
+    Returns:
+            Boolean: True if it was successful and False otherwise. False could mean that there was
+                     an error or that you were waiting for the operation to finish and it timed out.
+    """
+    folder_obj = self.get_folder(folder, collection=collection)
```
Comment (jahlbornG) [inline guru/core.py:2637]: i don't think you want to pass in `collection` here, since it is the dest collection.

File: guru/core.py [hunk 2]
```diff
@@ -2614,6 +2614,64 @@ def add_card_to_folder(self, card, target_folder):
...
+
+    collection_obj = self.get_collection(collection)
+    if not collection_obj:
+      raise ValueError(f"could not find collection: {collection}")
+
+    # if the folder is already in that collection, do nothing.
+    if folder_obj.collection and folder_obj.collection.id == collection_obj.id:
+      self.__log(make_red("folder", folder_obj.title,
+                          "is already in collection", collection_obj.name))
+      return
```
Comment (jahlbornG) [inline guru/core.py:2649]: should this return a value?

Comment (markhornak) [reply guru/core.py:2649]: yes, should return False.

File: guru/core.py (hunk 1 above)
Comment (markhornak) [reply guru/core.py]: fixed here: [2e33159](https://github.com/guruhq/py-sdk/pull/58/commits/2e33159cc5e3732d41c999b385bf36582b271102)

**Pull request merged**